framework/Common.h

```diff
@@ -165,20 +165,20 @@ class idCommon {
 	virtual void				SetRefreshOnPrint( bool set ) = 0;
 
 								// Prints message to the console, which may cause a screen update if com_refreshOnPrint is set.
-	virtual void				Printf( const char *fmt, ... )id_attribute((format(printf,2,3))) = 0;
```

Comment:
why do you remove the attribute?

Reply:
Without 86872eedc291e69c3ba41119b3ab8aa368ceb89f (the attribute removal), the compile fails with:

```
renderer/Model_ma.cpp:700:20: error: format not a string literal and no format arguments [-Werror=format-security]
  700 |         MA_VERBOSE((va("MESH %s - parent %s\n", header.name, header.parent)));
      |
```

Reply:
(...and after that, if I remember correctly, I noticed that `dhewm3` does _not_ include those attributes.. so I removed them, and it worked.  not hugely scientific)

Reply:
Of course when you look at the code more closely (I didn't either, initially) you'll see that MA_VERBOSE is only used in one place and the proper solution is to just get rid of MA_VERBOSE and modify that one place (in end of MA_ParseMesh()) like this:
```c++
	if(maGlobal.verbose) {
		common->Printf("MESH %s - parent %s\n", header.name, header.parent);
		common->Printf("\tverts:%d\n",maGlobal.currentObject->mesh.numVertexes);
		common->Printf("\tfaces:%d\n",maGlobal.currentObject->mesh.numFaces);
	}
```

Reply:
> I'll push some more commits soon, but after reverting https://github.com/blendogames/quadrilateralcowboy/commit/86872eedc291e69c3ba41119b3ab8aa368ceb89f and removing the macro, I'm running into:

Ignore this.. your explanation about the `id_attribute` removals hiding these errors fits as the reason.  A few more (simple) updates required to fix these.

Reply:
yes, using va() in printf-like functions makes no sense - the whole point of va() is that it provides an easy way to use printf-formatting with functions that take a normal string

I guess the instances where Doom uses both at the same time somehow grew organically during development, maybe common->Warning() and the likes *did* only take a simple `const char*` at some point..

Reply:
oh, and it turns out that while dhewm3 *does* have those annotations, it suppresses format security warnings: https://github.com/dhewm/dhewm3/blob/a7e4eb86449799379f1e62b8a9890abab8fa457f/neo/CMakeLists.txt#L297-L300

I should fix that eventually

Reply:
Yes, that would be great! :)

Reply:
Oops; noticed while looking for something else that I omitted the conditional check around the model-loading (verbose) debug output.  Restoring that soon.